docs/integrating-uptest-for-e2e-testing.md

```diff
@@ -0,0 +1,143 @@
+2. Clone your demo repository on your local and `cd` into the root directory.
+3. Initialize build submodule with
+
+	```bash
+	make submodules
+	```
+
+4. First we will add a simple setup script that will deploy a secret and a provider config for our provider. 
+
+	```bash
+	mkdir -p cluster/test
+	touch cluster/test/setup.sh
+	chmod +x cluster/test/setup.sh
+
+	cat <<EOF > cluster/test/setup.sh
+	#!/usr/bin/env bash
+	set -aeuo pipefail
+
+	echo "Running setup.sh"
+	echo "Creating cloud credential secret..."
+	\${KUBECTL} -n upbound-system create secret generic provider-secret --from-literal=credentials="{\"token\":\"\${UPTEST_CLOUD_CREDENTIALS}\"}" \
+	--dry-run=client -o yaml | \${KUBECTL} apply -f -
+
+	echo "Waiting until provider is healthy..."
+	\${KUBECTL} wait provider.pkg --all --for condition=Healthy --timeout 5m
+
+	echo "Waiting for all pods to come online..."
+	\${KUBECTL} -n upbound-system wait --for=condition=Available deployment --all --timeout=5m
+
+	echo "Creating a default provider config..."
+	cat <<EOF | \${KUBECTL} apply -f -
+	apiVersion: github.upbound.io/v1beta1
+	kind: ProviderConfig
+	metadata:
+	  name: default
+	spec:
+	  credentials:
+	    source: Secret
+	    secretRef:
+	      name: provider-secret
+	      namespace: upbound-system
+	      key: credentials
+	EOF
+	```
+
+5. Now, please add the following lines to your Makefile which which will add `uptest` and `e2e` targets:
+
+	```Makefile
+	# ====================================================================================
+	# End to End Testing
+	CROSSPLANE_NAMESPACE = upbound-system
+	-include build/makelib/local.xpkg.mk
+	-include build/makelib/controlplane.mk
+
+	uptest: $(UPTEST) $(KUBECTL) $(KUTTL)
+		@$(INFO) running automated tests
+		@KUBECTL=$(KUBECTL) KUTTL=$(KUTTL) $(UPTEST) e2e "${UPTEST_EXAMPLE_LIST}" --setup-script=cluster/test/setup.sh || $(FAIL)
+		@$(OK) running automated tests
+
+	e2e: build controlplane.up local.xpkg.deploy.provider.$(PROJECT_NAME) uptest
+	```
+
+## Testing Locally
+
+1. Generate a [Personal Access Token](https://github.com/settings/tokens/new) for your Github account with
+    `repo/public_repo` and `delete_repo` scopes.
+2. Run the following:
+
+	```bash
+	export UPTEST_CLOUD_CREDENTIALS=<your-token-here>
+	UPTEST_EXAMPLE_LIST=examples/repository/repository.yaml make e2e
+	```
+
+You should see a `PASS` at the end of logs indicating everything worked fine.
+
+## Adding the GitHub workflow
+
+Now we have things working locally, let's add a GitHub workflow to automate end to end testing with CI.
+
+1. Run the following to add the GitHub workflow definition which will be triggered for `issue_comment` events and will call
+uptests reusable workflow:
+
+	```bash
+	cat <<EOF > .github/workflows/e2e.yaml
+	name: End to End Testing
+
+	on:
+	  issue_comment:
+	    types: [created]
+
+	jobs:
+	  e2e:
+	    uses: upbound/uptest/.github/workflows/pr-comment-trigger.yml@main
```

Comment:
@ytsarev if you want to give it a try before this PR merges, you would need to replace this line as below:

```suggestion
	    uses: turkenh/uptest/.github/workflows/pr-comment-trigger.yml@main
```